assets/src/dashboard/utils/cssLerp.js

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const cssLerp = (start, end, progress) => {
+  return `calc(calc(calc(1 - var(${progress}, 0)) * ${start}) + calc(var(${progress}, 0) * ${end}))`;
```

Comment:
Let's plan on adding a unit test here to make sure this linear interpolator always returns the expected values.

Reply:
Super nit, just to make this function easier to read do you think it'd be beneficial to re-write it to something like this:
```
const remainingProgress = `calc(1 - var(${progress}, 0))`;
return `
  calc(
    calc(${remainingProgress} * ${start}) +
    calc(var(${progress}, 0) * ${end})
)`;
```

It makes it a little easier to read that this whole thing is basically the sum of two sub `calcs` 🤔

Reply:
You could even added:
```
const currentProgress = `var(${progress}, 0)`;
```
And then the logic would be really clear to read.